packages/backend/src/accounting/psql/balance.test.ts

```diff
@@ -0,0 +1,186 @@
+import { createTestApp, TestContainer } from '../../tests/app'
+import { Config } from '../../config/app'
+import { initIocContainer } from '../../'
+import { Asset } from '../../asset/model'
+import { randomAsset } from '../../tests/asset'
+import { truncateTables } from '../../tests/tableManager'
+import { LedgerAccount } from './ledger-account/model'
+import { createLedgerAccount } from '../../tests/ledgerAccount'
+import { getAccountBalances } from './balance'
+import { ServiceDependencies } from './service'
+import { createLedgerTransfer } from '../../tests/ledgerTransfer'
+import { LedgerTransferState } from './ledger-transfer/model'
+
+describe('Balances', (): void => {
+  let serviceDeps: ServiceDependencies
+  let appContainer: TestContainer
+  let knex: Knex
+  let asset: Asset
+
+  beforeAll(async (): Promise<void> => {
+    const deps = initIocContainer({ ...Config, useTigerbeetle: false })
+    appContainer = await createTestApp(deps)
+    serviceDeps = {
+      logger: await deps.use('logger'),
+      knex: await deps.use('knex')
+    }
+    knex = appContainer.knex
+  })
+
+  beforeEach(async (): Promise<void> => {
+    asset = await Asset.query().insertAndFetch(randomAsset())
+  })
+
+  afterEach(async (): Promise<void> => {
+    jest.useRealTimers()
+    await truncateTables(knex)
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.shutdown()
+  })
+
+  describe('getAccountBalances', (): void => {
+    let account: LedgerAccount
+    let peerAccount: LedgerAccount
+
+    beforeEach(async (): Promise<void> => {
+      asset = await Asset.query(knex).insertAndFetch(randomAsset())
+      ;[account, peerAccount] = await Promise.all([
+        createLedgerAccount({ ledger: asset.ledger }, knex),
+        createLedgerAccount({ ledger: asset.ledger }, knex)
+      ])
+    })
+
+    test('gets balances for account without transfers', async (): Promise<void> => {
+      await expect(getAccountBalances(serviceDeps, account)).resolves.toEqual({
+        creditsPosted: 0n,
+        creditsPending: 0n,
+        debitsPosted: 0n,
+        debitsPending: 0n
+      })
+    })
+
+    test('ignores voided transfers', async (): Promise<void> => {
+      await createLedgerTransfer(
+        {
+          ledger: account.ledger,
+          creditAccountId: account.id,
+          debitAccountId: peerAccount.id,
+          state: LedgerTransferState.VOIDED,
+          amount: 10n
+        },
+        knex
+      )
+
+      await expect(getAccountBalances(serviceDeps, account)).resolves.toEqual({
+        creditsPosted: 0n,
+        creditsPending: 0n,
+        debitsPosted: 0n,
+        debitsPending: 0n
+      })
+    })
+
+    describe('calculates balances for single transfers', (): void => {
+      const amounts = {
+        credit: {
+          POSTED: 40n,
+          PENDING: 30n
+        },
+        debit: {
+          POSTED: 20n,
+          PENDING: 10n
+        }
+      }
+
+      for (const type of ['credit', 'debit']) {
```

Comment:
I think some these tests would fail if we started enforcing balance limits
https://github.com/interledger/rafiki/pull/1111#discussion_r1105683510

Reply:
Since I'm updating the `createTransfers` method to do balance checks itself, I don't think these tests should be converted to use the actual `createTransfers` method, but continue on creating transfers directly with the model.

TL;DR, I don't think these tests should necessarily care/need to know about balance checks.

Reply:
:thinking: how bad would it be to enforce balance limits in 
https://github.com/interledger/rafiki/blob/b9719023dff1d585667145bf899bcc73e9a40143/packages/backend/src/accounting/psql/ledger-transfer/model.ts#L39-L42

Reply:
🤔 
Should be a relatively straightforward change, would just need to have the balance calculation live as a method on the `LedgerTransfer` model... it does prevent errors _if_ someone were to use the model directly to create transfers, but seems a bit hidden at first to see where the balance checks are done, makes serving up errors a bit more convoluted.

Thinking if it's worth it...